src/AzureClient/EntryPoint/EntryPoint.cs

```diff
@@ -115,9 +115,62 @@ private ArgumentValue GetArgumentValue(System.Reflection.ParameterInfo parameter
+                        .ToImmutableArray<ArgumentValue>();
+                    return ArgumentValue.Array.TryCreate(values, ArgumentType.Bool) ??
+                        throw new ArgumentException($"Could not create array of Bool for {parameter.Name}.");
+                }
+                else if (arrayType == typeof(double))
+                {
+                    var values = Newtonsoft.Json.JsonConvert.DeserializeObject<double[]>(parameterValue)
+                        .Select(v => new ArgumentValue.Double(v))
+                        .ToImmutableArray<ArgumentValue>();
+                    return ArgumentValue.Array.TryCreate(values, ArgumentType.Double) ??
+                        throw new ArgumentException($"Could not create array of Bool for {parameter.Name}.");
+                }
+                else if (arrayType == typeof(long))
+                {
+                    var values = Newtonsoft.Json.JsonConvert.DeserializeObject<long[]>(parameterValue)
+                        .Select(v => new ArgumentValue.Int(v))
+                        .ToImmutableArray<ArgumentValue>();
+                    return ArgumentValue.Array.TryCreate(values, ArgumentType.Int) ??
+                        throw new ArgumentException($"Could not create array of Int for {parameter.Name}.");
+                }
+                else if (arrayType == typeof(string)){
+                    var values = Newtonsoft.Json.JsonConvert.DeserializeObject<string[]>(parameterValue)
+                        .Select(v => new ArgumentValue.String(v))
+                        .ToImmutableArray<ArgumentValue>();
+                    return ArgumentValue.Array.TryCreate(values, ArgumentType.String) ??
+                        throw new ArgumentException($"Could not create array of String for {parameter.Name}.");
+                }
+                else if (arrayType == typeof(Pauli))
+                {
+                    var values = Newtonsoft.Json.JsonConvert.DeserializeObject<Pauli[]>(parameterValue)
+                        .Select(v => new ArgumentValue.Pauli(v))
+                        .ToImmutableArray<ArgumentValue>();
+                    return ArgumentValue.Array.TryCreate(values, ArgumentType.Pauli) ??
+                        throw new ArgumentException($"Could not create array of Pauli for {parameter.Name}.");
+                }
+                else if (arrayType == typeof(Result))
+                {
+                    var values = Newtonsoft.Json.JsonConvert.DeserializeObject<Result[]>(parameterValue)
+                        .Select(v => new ArgumentValue.Result(v))
+                        .ToImmutableArray<ArgumentValue>();
+                    return ArgumentValue.Array.TryCreate(values, ArgumentType.Result) ??
+                        throw new ArgumentException($"Could not create array of Result for {parameter.Name}.");
+                }
+
+                throw new ArgumentException($"Invalid array type {arrayType}.");
```

Comment:
This error message seems to imply that an array (for instance) of type `Double[][]` is invalid rather than unsupported for use as a parameter to an entry point. That is, `Double[][]` is a valid type even if not supported in that context.

Reply:
Change the message to state that it is unsupported rather than invalid.